### PR TITLE
Upgrade text component interpolation features

### PR DESCRIPTION
**Pull request by TPRobots**
Pull request opened by github-pullrequestcreator.

### CODE REVIEW

File: src/text.tsx [hunk 1]
```diff
@@ -6,23 +6,90 @@ export interface Tag {
...
+const nodeRegex = (begin: string, end: string) =>
+  new RegExp(`${escapeRegex(begin)}(.*)${escapeRegex(end)}`, 'g');
+
+const hasKeys = (obj: Interpolations) => Object.keys(obj).length > 0;
+
+export const Text = ({
+  id,
+  interpolations = {},
+  tag = { start: '[', end: ']' },
+  suffix = { begin: '-begin', end: '-end' },
```
Comment (madsgodvinjensen) [inline src/text.tsx:50]: is the matching case sensitive? Does `[LINK-BEGIN]` match a string that includes `[link-begin]`?

**Review by madsgodvinjensen — APPROVED** [review]

File: src/text.tsx [hunk 2]
```diff
@@ -6,23 +6,90 @@ export interface Tag {
...
 }
 
-export const Text = ({ id, interpolations, tag }: TextProps) => {
+const toTag = (key: string, tag: Tag, suffix: string = '') =>
+  `${tag.start}${key}${suffix}${tag.end}`;
+
+const escapeRegex = (str: string) => str.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1');
+
+const nodeRegex = (begin: string, end: string) =>
+  new RegExp(`${escapeRegex(begin)}(.*)${escapeRegex(end)}`, 'g');
```
Comment (madsgodvinjensen) [inline src/text.tsx:42]:
```suggestion
  new RegExp(`${escapeRegex(begin)}(.*)${escapeRegex(end)}`, 'gi');
```

Would this make it case-insensitive?

**Review by madsgodvinjensen — CHANGES_REQUESTED** [review]

File: src/text.tsx (hunk 2 above)
Comment (NicoJanvier) [reply src/text.tsx]: Needed an additional change 😄

File: src/text.tsx [hunk 3]
```diff
@@ -6,23 +6,90 @@ export interface Tag {
   end: string;
 }
 
-export interface Interpolations {
+export interface Suffix {
+  begin: string;
+  end: string;
+}
+
+interface SingleInterpolations {
```
[the remark this answers is not in the document]
Comment (NicoJanvier) [reply src/text.tsx:14]: I am using `SingleInterpolations` and `PairInterpolations` further down this file.

**Review by madsgodvinjensen — APPROVED** [review]

**Pull request merged**